### ISSUE TITLE
onremotestream called twice for each remote stream

### ISSUE THREAD

**Issue by imaustink** [meetecho/janus-gateway#1313]
It seems that the `ontrack` event listener on instances of `RTCPeerConnection` is being called twice for each peer connection added to the session.

I first observed this behavior in my application and confirmed it is consistent with the [videoroom demo](https://janus.conf.meetecho.com/videoroomtest.html).

Since the ontrack handler is native funcationality, that leads me to beleive this is caused by Janus sending the same stream multiple times.

__Steps to reproduce:__
1. Open the videoroom demo
2. Put a break point [here](https://github.com/meetecho/janus-gateway/blob/master/html/videoroomtest.js#L552)
3. Join the room with a remote peer already in the room

**Comment by imaustink** [comment]
Upon further inspection, it seems that `ontrack` is called once for `audio` and again for `video`, which seems alright. However, this results in `onremotestream` getting called twice twice with the same `MediaStream`, which seems wrong to me.

**Comment by lminiero** [comment]
Actually it's not an issue or bug, but a direct consequence of when we introduced `ontrack` support in #942. You can find an explanation in the description there. All the demos handle this, so you can refer to those to see how to do it in your application instead. Closing.

**Comment by imaustink** [comment]
Thanks for the quick response, @lminiero!

While I understand this isn't strictly speaking a "bug", I think it makes consuming remote streams just a bit clumsy. I am aware this is on the edge of OCD, but anytime I see an opportunity to make an interface cleaner I will always opt for it, especially in the open source world.

I would be happy to submit a PR to change this behavior, or, at the very least, submit a PR to document this behavior. If you are open to either, please let me know and I would jump at the opportunity to assist.

**Comment by lminiero** [comment]
Since this would mean messing with MediaStream instances, both in janus.js and in the application itself, I think it's not a good idea and would rather keep it the way it is. Besides, how would I know that the fact I only get video the second time means that this is to be added to the video I got before, and not that audio has been removed? I don't think the way it works now is broken, but should rather be seen as updates to what the current stream is.
